core/runtime/TRTEngine.cpp

```diff
@@ -120,16 +120,25 @@ TRTEngine::TRTEngine(
   } else {
     uint64_t inputs_size = _in_binding_names.size();
     in_binding_names.resize(inputs_size);
-    for (size_t pyt_idx = 0; pyt_idx < inputs_size; pyt_idx++) {
+    for (uint64_t pyt_idx = 0; pyt_idx < inputs_size; pyt_idx++) {
       auto binding_name = _in_binding_names[pyt_idx];
-      auto trt_idx = cuda_engine->getBindingIndex(binding_name.c_str());
-      std::string engine_binded_name = cuda_engine->getIOTensorName(trt_idx);
-      TORCHTRT_CHECK(
-          (binding_name == engine_binded_name),
-          "Could not find a TensorRT engine binding for input named " << binding_name);
+      // Check if the binding name provided is in the list of engine's bindings
+      // by iterating through nbIOTensors and verify it is an input binding
+      bool is_binding = false, is_input = false;
+      int32_t trt_idx;
+      for (int32_t idx = 0; idx < cuda_engine->getNbIOTensors(); idx++) {
+        std::string curr_bind_name = cuda_engine->getIOTensorName(idx);
+        if (curr_bind_name == binding_name) {
+          is_binding = true;
+          trt_idx = idx;
+          if (cuda_engine->getTensorIOMode(binding_name.c_str()) == nvinfer1::TensorIOMode::kINPUT) {
+            is_input = true;
```

Comment:
Minor comment, but would it be safe to `break` here? Or maybe outside the innermost `if`? It seems once the binding name is found, the loop can safely stop